zerver/views/video_calls.py

```diff
@@ -151,3 +153,35 @@ def deauthorize_zoom_user(request: HttpRequest) -> HttpResponse:
             auth=(settings.VIDEO_ZOOM_CLIENT_ID, settings.VIDEO_ZOOM_CLIENT_SECRET),
         ).raise_for_status()
     return json_success()
+
+
+@zulip_login_required
+@never_cache
+def get_bbb_url(request: HttpRequest, user_profile: UserProfile) -> HttpResponse:
+    id = "zulip-" + str(random.randint(100000000000, 999999999999))
+    password = ''.join(random.SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in range(10))
+    checksum = hashlib.sha1(("create" + "meetingID=" + id + "&moderatorPW=" + password + settings.BIG_BLUE_BUTTON_SECRET).encode()).hexdigest()
+    return json_success({"url": "/calls/bbb/join?meeting_id=%22" + id + "%22&password=%22" + password + "%22&checksum=%22" + checksum + "%22"})
+
+
+@zulip_login_required
+@never_cache
+@has_request_variables
+def join_bbb(request: HttpRequest, meeting_id: str = REQ(validator=check_string), password: str = REQ(validator=check_string), checksum: str = REQ(validator=check_string)) -> HttpResponse:
+    if settings.BIG_BLUE_BUTTON_URL is not None and settings.BIG_BLUE_BUTTON_SECRET is not None:
+        create_params = "meetingID=" + meeting_id + "&moderatorPW=" + password
+        response = requests.get(
+            settings.BIG_BLUE_BUTTON_URL + "api/create?" + create_params + "&checksum=" + checksum)
+        try:
+            response.raise_for_status()
+        except Exception:
+            return json_error(("Connection error to Big Blue Button server"))
+        if "<messageKey>checksumError</messageKey>" in response.text:
+            return json_error(("Authentication to Big Blue Button can not established"))
+        join_params = "meetingID=" + meeting_id + "&password=" + password + "&fullName=" + quote(
+            request.user.full_name)
+        return redirect(settings.BIG_BLUE_BUTTON_URL + "api/join?" + join_params + "&checksum="
+                        + hashlib.sha1(("join" + join_params + settings.BIG_BLUE_BUTTON_SECRET)
+                                       .encode()).hexdigest())
```

Comment:
I think it'd be a lot more readable to generate all there parameters using `urllib` rather than concatenating strings.

Or maybe the functions in `zerver.lib.url_encoding`.